src/XRP/issued-currency-client.ts

```diff
@@ -860,4 +911,34 @@ export default class IssuedCurrencyClient {
       sender,
     )
   }
+
+  /**
+   * Calculates the sendMaxValue by applying the transferFee to the amount being sent and ensuring the total amount
+   * fits within maximum decimal precision.
+   * @see https://xrpl.org/currency-formats.html#issued-currency-precision
+   *
+   * @param amount The amount of issued currency to pay to the destination.
+   * @param transferFee The transfer fee associated with the issuing account, expressed as a percentage.
+   *                    (i.e. a value of .5 indicates a 0.5% transfer fee).
+   */
+  public calculateSendMaxValue(amount: string, transferFee: number): string {
```

Comment:
NICE, this is far easier to understand

one minor architecture thing: it's a bit strange that `IssuedCurrencyClient` exposes `calculateSendMaxValue`, since it doesn't seem like something that a caller would care about. i'm guessing we're doing this so that it's testable.

i think this is fine, but one way you could fix this is by making a module called something like `IssuedCurrencyUtil` that has this function in it. that way, you can still test it, but it keeps the abstraction tighter. the downside is another level of indirection; the upside is that IssuedCurrencyClient becomes much dumber

in general, you don't really want to test private functions because they tend to be implementation details, and you're often better off testing them indirectly via the public interface. that being said, you could also make this function private and use `@ts-ignore` to ignore the private typecheck. types aren't "real" anyway

Reply:
yeah so... I actually thought about these considerations as well, and the reason I decided to just make it simple and leave it public is:  I think it may likely be the case that a user of the SDK will actually want to be able to calculate the sendMax value separately from this `sendIssuedCurrencyPayment` method in order to decide if it's viable/cheap enough, before actually submitting the transaction...  So I think for now I want to leave it public bc I can actually see us using this in our demo, and if we end up not needing it or can think of a better way to obscure it (or feel that it makes more sense in a utility class alongside maybe some other stuff we end up writing) I think we should move it then.